doc/source/serve/develop-and-deploy.md

```diff
@@ -0,0 +1,138 @@
+  inside a dictionary formatted as `[{"translation_text": "..."}]`.
+- The `Translator`'s `translate` method extracts the translated text by indexing into the dictionary.
+
+You can copy-paste this script and run it locally. It translates `"Hello world!"`
+into `"Bonjour Monde!"`.
+
+```console
+$ python model.py
+
+Bonjour Monde!
+```
+
+Let's convert this model into a Ray Serve application with FastAPI. For other HTTP options, please look at [Set Up FastAPI and HTTP](serve-set-up-fastapi-http). These are the 3 changes we need:
+1. Import Ray Serve and Fast API dependencies
+2. Add decorators for serve deployment with FastAPI - `@serve.deployment` and `@serve.ingress(app)`
+3. `bind` our `Translator` deployment to arguments that will be passed into its constructor
+
+```{literalinclude} ../serve/doc_code/develop_and_deploy/model_deployment_with_fastapi.py
+:start-after: __deployment_start__
+:end-before: __deployment_end__
+:language: python
+```
+
+You will notice that there are properties being configured for the deployment, such as `num_replicas` and `ray_actor_options`. These help configure the number of copies of our deployment and the resource requirements per copy. For a complete guide on the configurable parameters on a deployment, please check out [configuring a serve deployment](serve-configure-deployment).
+
+## Testing a Ray Serve application locally
+
+To test locally, we run the script with the `serve run` CLI command. This command takes in an import path formatted as `module:application`. Make sure to run the command from a directory containing a local copy of this script saved as `model.py`, so it can import the application:
+
+```console
+$ serve run model:translator_app
+```
+
+This command will run the `translator_app` application and then block, streaming logs to the console. It can be killed with `Ctrl-C`, which will tear down the application.
+
+We can now test our model over HTTP. It can be reached at the following URL by default:
+
+```
+http://127.0.0.1:8000/
+```
+
+We'll send a POST request with JSON data containing our English text. Here's a client script that requests a translation for "Hello world!":
+
+```{literalinclude} ../serve/doc_code/develop_and_deploy/model_deployment_with_fastapi.py
+:start-after: __client_function_start__
+:end-before: __client_function_end__
+:language: python
+```
+
+While a Ray Serve application is deployed, you can use the `serve status` CLI command to check the status of the applications and deployments. For more details on the output format of `serve status`, please look at [Inspecting Serve in production guide](serve-in-production-inspecting).
+
+```console
+$ serve status
+name: default
+app_status:
+  status: RUNNING
+  message: ''
+  deployment_timestamp: 1687415211.531879
+deployment_statuses:
+- name: default_Translator
+  status: HEALTHY
+  message: ''
+```
+
+## Building Serve config files for production deployment
+
```

Comment:
Suggest a sentence or two about why I need to create a Serve config file.

Reply:
@edoakes any suggestions for how to phrase the reasons?

Reply:
"A Serve config file is the single source of truth for the cluster, allowing you to specify system-level configuration and your applications in one place. It also allows you to declaratively update your applications."